crates/interpreter/src/valid.rs

```diff
@@ -129,13 +128,21 @@ impl<'m> Context<'m> {
         let mut side_tables = vec![];
         if let Some(mut parser) = self.check_section(parser, SectionId::Code)? {
             check(self.funcs.len() == imported_funcs + parser.parse_vec()?)?;
+            let mut offset = 0;
             for x in imported_funcs .. self.funcs.len() {
                 let size = parser.parse_u32()? as usize;
+                offset += size_of::<u32>();
                 let mut parser = parser.split_at(size)?;
-                let t = self.functype(x as FuncIdx).unwrap();
+                let t = self.functype(x as FuncIdx)?;
```

Comment:
This call can't fail which is why it's using `unwrap()`. We check `funcs.len()` above and we also have a check against `types.len()` when pushing to `funcs`.